Compost - Exception, right click on full barrel

### Minecolonies version

of latest branch of 1.12 7.15.18

### Expected behavior

-  Not sure.  I've not seen any documentation on this feature

### Actual behaviour

When I right click on a barrel with "Spruce saplings" the barrel became full and a lid appeared on the barrel.
Then I right clicked on it again.  Trying to remove item(I think).  I got an exception.

Based on the code,  right clicking a second time tries to put more items in the compost.   There is not check to make sure the item clicking is a good ore.

00:20:41.667 [Server thread] FATAL net.minecraft.server.MinecraftServer - Error executing task
java.util.concurrent.ExecutionException: java.lang.IllegalArgumentException: Stack can not be invalid!
	at java.util.concurrent.FutureTask.report(Unknown Source) ~[?:1.8.0_144]
	at java.util.concurrent.FutureTask.get(Unknown Source) ~[?:1.8.0_144]
	at net.minecraft.util.Util.runTask(SourceFile:47) [Util.class:?]
	at net.minecraft.server.MinecraftServer.updateTimeLightAndEntities(MinecraftServer.java:723) [MinecraftServer.class:?]
	at net.minecraft.server.MinecraftServer.tick(MinecraftServer.java:668) [MinecraftServer.class:?]
	at net.minecraft.server.integrated.IntegratedServer.tick(IntegratedServer.java:185) [IntegratedServer.class:?]
	at net.minecraft.server.MinecraftServer.run(MinecraftServer.java:526) [MinecraftServer.class:?]
	at java.lang.Thread.run(Unknown Source) [?:1.8.0_144]
Caused by: java.lang.IllegalArgumentException: Stack can not be invalid!
	at net.minecraftforge.oredict.OreDictionary.getOreIDs(OreDictionary.java:493) ~[OreDictionary.class:?]
	at com.minecolonies.coremod.tileentities.TileEntityBarrel.checkCorrectItem(TileEntityBarrel.java:166) ~[TileEntityBarrel.class:?]
	at com.minecolonies.coremod.tileentities.TileEntityBarrel.useBarrel(TileEntityBarrel.java:119) ~[TileEntityBarrel.class:?]
	at com.minecolonies.coremod.blocks.BlockBarrel.onBlockActivated(BlockBarrel.java:159) ~[BlockBarrel.class:?]
	at net.minecraft.server.management.PlayerInteractionManager.processRightClickBlock(PlayerInteractionManager.java:446) ~[PlayerInteractionManager.class:?]
	at net.minecraft.network.NetHandlerPlayServer.processTryUseItemOnBlock(NetHandlerPlayServer.java:739) ~[NetHandlerPlayServer.class:?]
	at net.minecraft.network.play.client.CPacketPlayerTryUseItemOnBlock.processPacket(SourceFile:55) ~[CPacketPlayerTryUseItemOnBlock.class:?]
	at net.minecraft.network.play.client.CPacketPlayerTryUseItemOnBlock.processPacket(SourceFile:11) ~[CPacketPlayerTryUseItemOnBlock.class:?]
	at net.minecraft.network.PacketThreadUtil$1.run(SourceFile:13) ~[PacketThreadUtil$1.class:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Unknown Source) ~[?:1.8.0_144]
	at java.util.concurrent.FutureTask.run(Unknown Source) ~[?:1.8.0_144]
	at net.minecraft.util.Util.runTask(SourceFile:46) ~[Util.class:?]
	... 5 more

--- a/src/main/java/com/minecolonies/coremod/tileentities/TileEntityBarrel.java
+++ b/src/main/java/com/minecolonies/coremod/tileentities/TileEntityBarrel.java
@@ -157,6 +157,11 @@ public class TileEntityBarrel extends TileEntity implements ITickable
 
     private boolean checkCorrectItem(final ItemStack itemStack)
     {
+        if(itemStack.isEmpty())
+        {
+            return false;
+        }
+
         for(final String string : Configurations.gameplay.listOfCompostableItems)
         {
             if(itemStack.getItem().getRegistryName().toString().equals(string))

